app/controllers/healthcare_controller.rb

```diff
@@ -1,44 +1,81 @@
 class HealthcareController < ApplicationController
+  include Wicked::Wizard
+
+  steps :physical, :mental, :social, :allergies, :needs, :transport, :contact
+
   def show
-    step.form.tap(&:prepopulate!)
-    render_cell(:healthcare, step)
+    form.prepopulate!
+    run_form_validations
+    render html: healthcare_form_cell, layout: true
   end
 
   def update
-    if form.validate(params[step_name])
+    if params.key? 'needs_add_medication'
+      form.deserialize permitted_params
+      form.add_medication
+      flash[:form_data] = form.to_parameter_hash
+      flash[:no_validate] = 'true'
+      redirect_to wizard_path
+    elsif form.validate permitted_params
       form.save
-      redirect_to next_path
+      if params.key? 'save_and_view_profile' || end_of_wizard?
+        redirect_to profile_path(escort)
+      else
+        redirect_to next_wizard_path
+      end
     else
-      render_cell(:healthcare, step)
+      flash[:form_data] = permitted_params
+      redirect_to wizard_path
     end
   end
 
-  def update_and_redirect_to_profile
-    if form.validate(params[step_name])
-      form.save
-      redirect_to profile_path(escort)
-    else
-      render_cell(:healthcare, step)
-    end
-  end
+  private
 
-  def add_medication
-    step.form
-    step.form.deserialize(params[step_name])
-    step.form.add_medication
+  def healthcare_form_cell
+    cell = cell(:healthcare, form)
 
-    render_cell :healthcare, step
+    cell.form_title = form.class.name
+    cell.prev_path = previous_wizard_path if is_prev_step?
+    cell.next_path = next_wizard_path unless end_of_wizard?
+    cell.current_question = current_step_index + 1
+    cell.total_questions = wizard_steps.size
+    cell.template_name = form.class.name
+    cell.form_path = wizard_path
+
+    cell
   end
 
-  private
+  def permitted_params
```

Comment:
How about just having this as `params`, the permitting part should be handled by reform right?